src/user-support.c

```diff
@@ -102,10 +102,32 @@ void setup_user_data()
 
 	if (user_data == NULL)
 		return;
+
 	// Only support absolute paths.
 	if (user_data[0] != '/') {
 		die("user data directory must be an absolute path");
 	}
 
 	mkpath(user_data);
 }
+
+void setup_user_xdg_runtime_dir()
+{
+	const char *xdg_runtime_dir = getenv("XDG_RUNTIME_DIR");
+
+	if (xdg_runtime_dir == NULL)
+		return;
+	// Only support absolute paths.
+	if (xdg_runtime_dir[0] != '/') {
+		die("XDG_RUNTIME_DIR must be an absolute path");
+	}
+
+	errno = 0;
+	mkpath(xdg_runtime_dir);
```

Comment:
I made the apparmor changes as part of this PR. See src/snap-confine.apparmor.in.

Thanks for the review!